kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java

```diff
@@ -296,4 +311,45 @@ private Optional<Long> loadLatestTransactionVersion(Engine engine, String applic
 
     return Optional.empty();
   }
+
+  /**
+   * Retrieves a map of domainName to {@link DomainMetadata} from the log files.
+   *
+   * <p>Loading domain metadata requires an additional round of log replay so this is done lazily
+   * only when domain metadata is requested. We might want to merge this into {@link
+   * #loadTableProtocolAndMetadata}.
+   *
+   * @param engine The engine used to process the log files.
+   * @return A map where the keys are domain names and the values are the corresponding {@link
+   *     DomainMetadata} objects.
+   * @throws RuntimeException if an I/O error occurs while closing the iterator.
+   */
+  private Map<String, DomainMetadata> loadDomainMetadataMap(Engine engine) {
```

Comment:
I am wondering if this is the right way. Do we need to load all the DMs? Or we need to load just the specific DM for a domain name? For example in the txnVersion (just above) our usecase turned out be just loading one.

Reply:
I’m kind of concerned about efficiency if we load the DM for only a specific domain each time. If we need DMs for multiple domains within a single transaction, would it require a log replay round for each domain?

We will have a separate PR to optimize how we load domain metadata by loading it together with the protocol and metadata, instead of in a separate round. Perhaps we can defer this to that PR.

Reply:
> We will have a separate PR to optimize how we load domain metadata by loading it together with the protocol and metadata

@qiyuandong-db it seems there's some uncertainty as to whether this is the best approach. Can we have a brief sync on this? Or can we put together a very brief 1 decision doc to discuss the pros and cons? I think we should do that _before_ you spend the time putting up a PR -- what do you think?

Reply:
I think that sounds good and we should do that! I’ll reach out to set up a sync before we start on this.